src/connection.ts

```diff
@@ -191,11 +198,39 @@ export class Connection extends JSForceConnection {
     try {
       this.setApiVersion(await this.retrieveMaxApiVersion());
     } catch (err) {
+      if (err.name === ORG_NOT_FOUND_ERROR_NAME) {
+        throw err; // throws on DNS connection errors
+      }
       // Don't fail if we can't use the latest, just use the default
       this.logger.warn('Failed to set the latest API version:', err);
     }
   }
 
+  /**
+   * Verify that instance has a reachable DNS entry, otherwise will throw error
+   */
+  public async isResolvable(): Promise<boolean> {
+    if (!this.options.connectionOptions?.instanceUrl) {
+      throw new SfdxError('Connection has no instanceUrl', 'NoInstanceUrl', [
+        'Make sure the instanceUrl is set in your command or config',
+      ]);
+    }
+    const resolver = await MyDomainResolver.create({
+      url: new URL(this.options.connectionOptions.instanceUrl),
+      timeout: Duration.seconds(10),
```

Comment:
@shetzel is there a user controllable duration that can be applied here? I see env var `SFDX_DOMAIN_RETRY` but that seems specific to waits for Lightning domain resolution during scratch org create, so not sure that can reasonably be used here.

Reply:
That should not be used here--it'll delay results for no gain.  Newly created scratch orgs have a version so this code doesn't run.

I think "does DNS resolve?" is a more binary question where there's not a lot of value in retries.  I wanted the Resolver to not retry at all (that original zero-second version) but I don't think it supports that.

Reply:
I believe the polling code waits after trying, so if the intent is to only try once, then zero is appropriate.

Reply:
It didn't seem to work that way.  With `0` every org would come back with an error.

I've been using this https://github.com/lifeomic/attempt for a few years and have enjoyed its flexibility.

Reply:
👍 for 10 seconds then.